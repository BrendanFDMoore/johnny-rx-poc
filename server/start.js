const app = require('express')();
const http = require('http').Server(app);
const io = require('socket.io')(http);

const logger = (msg) => console.log('Server says:', msg);

io.on('connection', (socket) => {
  logger('a user connected!');
  socket.emit('message', 'greetings user!');
  socket.on('disconnect', () => {
    logger('user disconnected');
  });
  socket.on('chat message', (msg) => {
    logger('message: ' + msg);
    io.emit('chat message', msg);
  });

  socket.on('johnny event', (msg) => {
    logger('received johnny event: ' + msg);
    io.emit('johnny event', msg);
  });
});

http.listen(4444, () => {
  logger('listening on *:4444');
});
