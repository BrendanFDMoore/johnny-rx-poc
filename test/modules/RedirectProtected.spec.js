import React from 'react';
import { MemoryRouter as Router } from 'react-router-dom';
import { renderToString } from 'react-dom/server';
import RedirectProtected from 'modules/RedirectProtected';

describe('utils/RedirectProtected', () => {
  it('should redirect for unauthenticated access', () => {
    const render = renderToString(
      <Router location="/private">
        <RedirectProtected
          exact
          pattern="/private"
          component={() => (<div>PRIVATE</div>)}
          isAuthenticated={false}
        />
      </Router>);

    expect(render).toMatchSnapshot();
  });

  it('should allow navigation for authenticated access', () => {
    const render = renderToString(
      <Router location="/private">
        <RedirectProtected
          exact
          pattern="/private"
          component={() => (<div>PRIVATE</div>)}
          isAuthenticated={true}
        />
      </Router>
    );

    expect(render).toMatchSnapshot();
  });
});
