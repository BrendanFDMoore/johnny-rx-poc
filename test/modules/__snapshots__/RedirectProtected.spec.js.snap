// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`utils/RedirectProtected should allow navigation for authenticated access 1`] = `"<div data-reactroot=\\"\\" data-reactid=\\"1\\" data-react-checksum=\\"-363261579\\">PRIVATE</div>"`;

exports[`utils/RedirectProtected should redirect for unauthenticated access 1`] = `"<!-- react-empty: 1 -->"`;
